polars/polars-arrow/src/kernels/rolling/no_nulls/mean.rs

```diff
@@ -12,9 +12,9 @@ impl<
         T: NativeType + IsFloat + std::iter::Sum + AddAssign + SubAssign + Div<Output = T> + NumCast,
     > RollingAggWindowNoNulls<'a, T> for MeanWindow<'a, T>
 {
-    fn new(slice: &'a [T], start: usize, end: usize) -> Self {
+    fn new(slice: &'a [T], start: usize, end: usize, params: Option<RollingFnParams>) -> Self {
```

Comment:
I think this should be a `Box<dyn Any>`. Then every function can downcast to its specific arguments type. In this PR's case no function will do that except for `var` and `std`.

If other functions need different arguments they can pass their own type.

Reply:
Thank you for taking a look. I'm relatively new to Rust so what makes that better than doing it this way?
To be extra clear, the type definition is
```
pub enum RollingFnParams {
    RollingVarParams { ddof: u8 },
}```
and I figured other variants could be added later but that explicitly defining what each function required was clearer and could be handled by a match where appropriate while preventing arbitrary input from going to these functions.
With the way you're suggesting, would I make `RollingVarParams` and any future param sets their own types? Mostly I'm curious about what issues you see here so I can learn something. Is it just pointless to define a type this vague?

Reply:
On mobile:

Ddof doesn't make any sense to pass to other rolling function like a rolling sum. That's why I propose using dyn Any. Then every function can only accept its own parameter type.

Reply:
Thank you for replying so quickly.
In my limited understanding, dyn Any would have the same problems as what I'm doing but even worse. Nothing would stop someone from passing anything at all to a rolling function. With the way it is now, at least there are defined enum variants for what kind of parameters each rolling function should accept and they have to be set ahead of time. Yeah, you could pass MeanParams (if they existed) to variance or VarParams to mean but it's up to the individual kernel to decide what to do just like with dyn Any. Is it better because I can then give each window an associated type through it's trait and cast the Any to that directly beforehand?

The only alternative I can think of that doesn't leave it up to the window itself is to have a separate type for parameters to each rolling function and put a generic type parameter on `RollingOptions`. But even then I think the function parameter types would have to be related somehow so they can pass through `rolling_agg` and have the same signature. And I don't know if this would work and it seems complicated.

I'm not trying to give you a hard time here. Only trying to learn a little. It doesn't seem like that big of a change to do it with Any so I'm happy to make the change if that's what you want.

Reply:
The point is that the `Box<dyn Any>` is in our private API. Not what we expose publicly for the user facing API.

The users will pass `RollingVarArgs` or a `RollingMeanArgs` and we will convert it to a `Box<Any>`. As I said, I don't want arguments needed for variance being passed to `rolling_mean` that doesn't make much sense, so we need some dynamic typing, hence the `dyn Any`. Think of it as `**kwargs` in python, where a specific implemenation of a function knows which keyword arguments to take out of them.

Reply:
Rolling variance args shouldn't be in an enum with rolling quantile args. 

What I mean is this:

```rust
use std::any::Any;
use std::sync::Arc;

/// Public type
pub struct RollingVarOptions{
    ddof: u8
}

/// Public type
pub struct RollingMeanOptions{
    mean_args: usize
}

/// Public function
pub fn rolling_var(args: RollingVarOptions) {
    let args = Arc::new(args) as Arc<dyn Any>;
    rolling_var_private(args)
}

/// Public function
pub fn rolling_mean(args: RollingMeanOptions) {
    let args = Arc::new(args) as Arc<dyn Any>;
    rolling_var_private(args)
}

/// Private functions
fn rolling_mean_private(args: Arc<dyn Any>) {
    let args = args.downcast_ref::<RollingMeanOptions>().unwrap();
    // implementation work
}

/// Private functions
fn rolling_var_private(args: Arc<dyn Any>) {
    let args = args.downcast_ref::<RollingVarOptions>().unwrap();
    // implementation work
}
```

This should work perfectly and have clear single function structs for the end user. It also reduces binary bloat which is important in this crate.

Reply:
OK, it took a little fiddling around with `Any` but it seems to do what it's supposed to now.
A few tests started failing due to numerical precision issues. They're being compared exactly, seemingly as a hack around pytest not being able to handle `None` in numeric outputs (though it supports `NaN`...). I could either change the values to what they are now or change the tests to check `None` and numerical precision of floats separately.